subiquity/common/types.py

```diff
@@ -87,6 +102,7 @@ class ApplicationStatus:
     state: ApplicationState
     confirming_tty: str
     error: Optional[ErrorReportRef]
+    nonreportable_error: Optional[NonReportableError]
     cloud_init_ok: Optional[bool]
     interactive: Optional[bool]
     echo_syslog_id: str
```

Comment:
@d-loose I am proposing to change the response object for `/meta/status` to include a new field `nonreportable_error`, to represent errors subiquity doesn't automatically generate apport reports for. In the case that `state` is `ApplicationState.ERROR`, if the error results in an apport report then `error` will be populated with the reportref as before; otherwise, it will be null/none and `nonreportable_error` will be filled in with a `NonReportableError`. Does this seem okay from the u-d-b side of things? The one thing I'm concerned about is if there's any spots that always expect `error` to be not empty when `state == ApplicationState.ERROR`.

Reply:
Thanks for the heads-up! I don't see any problems with this on our side. Currently any `ApplicationState.ERROR` is treated as fatal to the installation process. The `error` isn't being used anywhere so far.

@spydon FYI